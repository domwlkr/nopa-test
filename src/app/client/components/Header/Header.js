import React, {PropTypes} from 'react';
import Button from '../Button/Button';
import * as Paths from '../../constants/paths';
import logo from '../../../static/images/Logo_Nopa.svg';

const Header = (props) => {
  const loginButton = !props.loggedIn ? <Button to={Paths.LOGIN_BANK}>Log In</Button> : <Button onClick={props.onLogoutClick}>Log Out</Button>;

  return (
    <div className="header">
      <img className="logo" src={logo} />
      <div className="login">
        {loginButton}
      </div>
    </div>
  );
};

Header.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  onLogoutClick: PropTypes.func.isRequired
};

export default Header;
