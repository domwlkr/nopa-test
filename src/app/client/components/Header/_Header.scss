.header {
  background: white;
  display: flex;
  justify-content: space-between;

  .logo {
    height: 55px;
    margin: 6px 0 5px 9px;
  }

  .login {
    margin: 9px 16px 11px 0;
  }
}

@media (min-width:380px) {
  .header {
    background-color: transparent;
    margin: 10px;
    flex-direction: row-reverse;

    .logo {
      display: none;
    }

    .login {
      align-self: flex-end;
    }
  }
}