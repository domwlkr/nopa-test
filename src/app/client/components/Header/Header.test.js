import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const loggedIn = true;
const bank = {name: 'Barclays', logo: 'test-file-stub'};
const onLogoutClick = jest.fn();

test('Renders correctly', () => {
  const header = renderer.create(
    <Header loggedIn={loggedIn} bank={bank} onLogoutClick={onLogoutClick} />
  ).toJSON();

  expect(header).toMatchSnapshot();
});

test('When login button is clicked event fired', () => {
  const header = shallow(
    <Header loggedIn={loggedIn} bank={bank} onLogoutClick={onLogoutClick} />
  );

  header.find('Button').simulate('click');

  expect(onLogoutClick).toHaveBeenCalled();
});