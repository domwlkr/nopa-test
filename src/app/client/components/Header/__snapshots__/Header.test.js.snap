// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="header"
>
  <img
    className="logo"
    src="test-file-stub"
  />
  <div
    className="login"
  >
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
    >
      Log Out
    </a>
  </div>
</div>
`;
